app/gui2/src/components/visualizations/TableVisualization.vue

```diff
@@ -265,12 +270,40 @@ function isMatrix(data: object): data is LegacyMatrix {
   return json.every((d) => d.length === firstLen)
 }
 
-function toField(name: string): ColDef {
-  return { field: name }
+function toField(name: string, valType?: string): ColDef {
+  let icon
+  switch (valType) {
+    case 'Char':
+      icon = 'text'
+      break
+    case 'Boolean':
+      icon = 'check'
+      break
+    case 'Integer':
+    case 'Float':
+      icon = 'math'
+      break
+    case 'Date':
+      icon = 'calendar'
+      break
+    case 'Mixed':
+      icon = 'mixed'
+  }
+  const svgTemplate = `<svg viewBox="0 0 16 16" width="16" height="16"> <use xlink:href="src/assets/icons.svg#${icon}"/> </svg>`
```

Comment:
This file path should not be hardcoded into the string here, since it will change during the build process. I'm honestly even surprised this worked in the first place, so maybe there is some support for this kind of declaration, but I'd still be more content with a proper `import`.

```ts
import icons from '@/assets/icons.svg'
```

This should give you `icons` variable that directly holds the path to the file. See how that is used in `SvgIcon.vue` as an example.